### ISSUE TITLE
Preview 4. Something goes wrong with cache.

### ISSUE THREAD

**Issue by dmitryhryppa** [HaxeFoundation/haxe#7195]
When compiling to JS, Haxe generates some unused stuff with `-dce full` flag in release mode. 
![Image](https://image.ibb.co/bW4dVJ/image.png)

But with `-debug` flag this code was generated as:
```
(function () { "use strict";
var $hxEnums = {};
var MyApp = function() { };
MyApp.main = function() {
	console.log("src/MyApp.hx:5:",123);
};
MyApp.main();
})();
```

**Comment by dmitryhryppa** [comment]
Oh, it's gone after restarting VSCode. This is weird... 
But now I understood what's going on. I had some enums in my class before. Then I removed them but Haxe continued to generate some stuff for them. After restarting VSCode problem is gone. 

Some time ago I got similar behaviour on cpp target. I removed a few Haxe classes and got linking error on compilation process. So, I cleared bin folder and started compilation again. Then, randomly, linking problem appeared again. I've checked generated sources and found that Haxe generated .h and .cpp files for classes that was removed from the project some time ago. 

So, looks like problem is in compilation server or vshaxe caching... Not sure.
Will try to reproduce it.
